### ISSUE TITLE
translation-match-params eslint rule errors with formatMessage.rich

### ISSUE THREAD

**Issue by bjohn465** [format-message/format-message#256]
When `formatMessage.rich` is used and the `format-message/translation-match-params` eslint rule is enabled, the eslint rule will complain about a missing placeholder in the translation string.

I created [a repository that demonstrates the problem](https://github.com/bjohn465/translation-match-params-example). The steps to reproduce are:
1. Clone [the repository](https://github.com/bjohn465/translation-match-params-example)
2. Run `yarn install`
3. Run `yarn eslint`

**Comment by vanwagonet** [comment]
Can you add a failing test to [translation-match-params.spec.js](https://github.com/format-message/format-message/blob/master/packages/eslint-plugin-format-message/__tests__/rules/translation-match-params.spec.js)?

**Comment by bjohn465** [comment]
I'll take a look and give it a try. Would you like that failing test as a pull request?

**Comment by vanwagonet** [comment]
yes please. If you can figure out a fix too, that be even better, but a failing test would be a great help.

**Comment by bjohn465** [comment]
I started looking into how this could be fixed, and here's what I've found so far:

- Adding the `tagsType` option to the `parse` call gets it most of the way there (see bjohn465/format-message@41cabdf4afd796223cf361747f07f80f7822eaf4)
- Changing `get-params-from-pattern-ast.js` to differentiate between a rich text param (e.g. `<b>`) and a regular param (e.g. `{b}`) is tricky

I gave changing `get-params-from-pattern-ast.js` a try with bjohn465/format-message@d2d15709eae06118b5a4e7c154626f6cb5f0c6d9, but I broke some other tests, likely because I didn't know about the `{a,<>,p{there}}` syntax that's used in `packages/format-message/__tests__/index.spec.js`.

`get-params-from-pattern-ast.js` would at least need to change to add children of the rich text params to the params array. The differentiation between rich text params and regular params would only be needed to catch cases in which a translation mixes up the two.
